V201/missing component and type enums #300

Jared-Newell-Mobility opened the pull request.

proelke commented on ocpp/v201/enums.py:
```diff
@@ -1349,3 +1349,566 @@ class SecurityEventType(str, Enum):
+    bytes_per_message = "BytesPerMessage"
+    enabled = "Enabled"
+    items_per_message = "ItemsPerMessage"
+    OfflineQueuingSeverity = "OfflineQueuingSeverity"
+    MonitoringBase = "MonitoringBase"
+    MonitoringLevel = "MonitoringLevel"
+    ActiveMonitoringBase = "ActiveMonitoringBase"
+    ActiveMonitoringLevel = "ActiveMonitoringLevel"
```
These aren't snake case.

Jared-Newell-Mobility replied:
Thanks - I've corrected this

proelke commented on ocpp/v201/enums.py:
```diff
@@ -1349,3 +1349,566 @@ class SecurityEventType(str, Enum):
+    Variable names where the component type is SmartChargingCtrlr
+    """
+
+    ACPhaseSwitchingSupported = "ACPhaseSwitchingSupported"
+    Available = "Available"
+    Enabled = "Enabled"
+    Entries = "Entries"
+    ExternalControlSignalsEnabled = "ExternalControlSignalsEnabled"
+    LimitChangeSignificance = "LimitChangeSignificance"
+    NotifyChargingLimitWithSchedules = "NotifyChargingLimitWithSchedules"
+    PeriodsPerSchedule = "PeriodsPerSchedule"
+    Phases3to1 = "Phases3to1"
+    ProfileStackLevel = "ProfileStackLevel"
+    RateUnit = "RateUnit"
```
These aren't snake case.

Jared-Newell-Mobility replied:
Thanks - I've corrected this

Jared-Newell-Mobility commented:
@OrangeTux regarding https://github.com/mobilityhouse/ocpp/pull/486#pullrequestreview-1679322180 - I used Appendices_CSV_v1.3 dm_components_vars.xlsx - this is referenced in OCPP 2.0.1 part 2 erratta v2.0 section 21.1 - where do you see the discrepancy?

OrangeTux reviewed (changes requested):
It's cumbersome task to add this enums, so thanks for the PR!

I didn't finish the review yet, since I found a few enums which are missing variants. That makes me wonder, based on which documents are the enums based? Are they based on the latest errata?

OrangeTux commented:
It seems that dm_components_vars.xlsx isn't matching the the written spec and the .csv files.

Jared-Newell-Mobility commented:
@OrangeTux - ok, so after contacting the OCA, I have updated this to break the components into logical and physical components, I have also extended the docstrings to document where the variables were sourced from. I have generic variables both as complete (this is to align with the documents provided in appendices_CSV_v1.3.zip, dm_components_vars.csv) and split out for each component. The components have all variables listed in both OCPP 2.0.1 Part 2 Appendices and dm_components_vars.csv, so this results in additional variables. 

Hopefully, this makes things a little more clear. 

In addition, there is some variable names such as selftest_active_set = "SelftestActive(Set)" and count_height_in_chars = "Count[HeightInChars]" - I am not sure if this is the best way to represent these. 

I didn't reply to the individual comments as they may no longer be relevant with the changes.

Looking forward to hear your thoughts.

tmh-grunwald-markus commented on ocpp/v201/enums.py:
```diff
@@ -1350,3 +1351,1347 @@ class SecurityEventType(str, Enum):
+
+    protocol_agreed = "ProtocolAgreed"
+    protocol_supported_by_ev = "ProtocolSupportedByEV"
+    vehicle_id = "VehicleID"
```
This is iffy. It's written `VehicleId` in the appendices PDF Section 3.2.13 and in variables.csv. Also all the other IDs like `EvseId` use the lower case `d`. But in `dm_components_vars.csv:116:ConnectedEV;VehicleID` it uses upper case...

Jared-Newell-Mobility replied:
Nice, catch this should be now corrected

tmh-grunwald-markus commented on ocpp/v201/enums.py:
```diff
@@ -1350,3 +1351,1347 @@ class SecurityEventType(str, Enum):
+    active = "Active"
+    dc_voltage = "DCVoltage"
+    enabled = "Enabled"
+    percent = "Percent"
```
In OCPP-2.0.1_part2_appendices_v13.pdf, 3.2.21. CPPWMController that's `Percentage`. I can't find a similar entry in the CSVs.

Jared-Newell-Mobility replied:
In this case, I thought it was a mistake - as the generic type is percent - it is only this one which calls it percentage ( in the description ) - I'll change it and if it is fixed in an Errata then it can be updated

tmh-grunwald-markus commented on ocpp/v201/enums.py:
```diff
@@ -1350,3 +1351,1347 @@ class SecurityEventType(str, Enum):
+    """
+
+    charging_station = "ChargingStation"
+    charging_station_identities = "ChargingStationIdentities"
+    charging_station_index = "ChargingStationIndex"
```
I can't find these entries, neither in OCPP-2.0.1_part2_appendices_v13.pdf 3.2.24. DistributionPanel (or the rest of the PDF) nor in the CSVs or the XSLX. Did I miss them? Where do they come from?

Jared-Newell-Mobility replied:
Corrected

tmh-grunwald-markus commented on ocpp/v201/enums.py:
```diff
@@ -1350,3 +1351,1347 @@ class SecurityEventType(str, Enum):
+    """
+
+    enabled = "Enabled"
+    enabled_set = "EnabledSet"
```
In OCPP-2.0.1_part2_appendices_v13.pdf, 3.2.53. TokenReader that's `Enabled(Set)`

Jared-Newell-Mobility replied:
Corrected

tmh-grunwald-markus commented on ocpp/v201/enums.py:
```diff
@@ -1350,3 +1351,1347 @@ class SecurityEventType(str, Enum):
+    See PhysicalComponentName for referenced physical component
+    """
+
+    active_set = "ActiveSet"
```
In In OCPP-2.0.1_part2_appendices_v13.pdf, 3.2.54. UpstreamProtectionTrigger that's `Active(Set)`.

Jared-Newell-Mobility replied:
Corrected

tmh-grunwald-markus commented on ocpp/v201/enums.py:
```diff
@@ -1350,3 +1351,1347 @@ class SecurityEventType(str, Enum):
+    Note: specific variables for each component are sourced from a union of
+    ocpp 2.0.1 part 2 appendices 3.2 v1.3 and dm_components_vars.csv
+    from appendices_CSV_v1.3.zip. That is for all Physical Components in
+    section 3.2. expressed in this module as enums.
+    e.g. the variables of ControllerVariableName enum
```
```suggestion
    section 3.2. expressed in this module as enums,
    e.g. the variables of ControllerVariableName enum
```
